### PR TITLE
Add tests of add_par() with empty DataFrames

### PR DESCRIPTION
**Pull request by khaeru**
## How to review

Note that CI checks pass, in particular newly added tests/parametrizations of `test_add_par` and `test_add_par2`.

## PR checklist

- [x] Add or expand tests.
  - [x] Test of an empty dataframe with correct columns.
  - [x] Test of an empty dataframe for a parameter with different idx_names/idx_sets, per https://github.com/iiasa/ixmp/issues/373#issuecomment-712056459
- ~Add, expand, or update documentation.~ N/A, bug fix
- [x] Update release notes.

### CODE REVIEW

File: ixmp/tests/core/test_scenario.py [hunk 1]
```diff
@@ -171,11 +171,26 @@ def test_scalar(self, scen):
...
+        scen = scen.clone(keep_solution=False)
+        scen.check_out()
+        scen.add_par(*args, **kwargs)
+
+    def test_add_par2(self, scen):
         scen = scen.clone(keep_solution=False)
         scen.check_out()
-        scen.add_par("b", ["new-york", "chicago"], value=100, unit="cases")
+        # Create a parameter with duplicate indices
+        scen.init_par("foo", idx_sets=["i", "i", "j"], idx_names=["i0", "i1", "j"])
```
Comment (khaeru) [inline ixmp/tests/core/test_scenario.py:192]: FYI @behnam-zakeri this new test gives me: `ValueError: Wrong number of items passed 3, placement implies 1`—same as you reported at https://github.com/iiasa/ixmp/issues/373#issuecomment-694195877.

So the error specifically occurs when the (a) the input data frame empty and (b) the parameter has ≥2 dimensions indexed by the same set, but with different names. Without (b) it doesn't occur.

With this pinned down, I can now look for a fix.
